### ISSUE TITLE
Rshiny gives error after loading rpart.object type using .RData file.

### ISSUE THREAD

**Issue by NxgChirag** [rstudio/shiny#945]
I have saved the .Rdata file, which contains 3 R objects:

    1. Vector
    2. Character String
    3. rpart.object (I figured out that this object in .RData only creating issue, as if I remove this object from .RData file, Shiny app works fine.)

Whenever I load the .RData file and refresh the application, Rshiny gives error below:

Error in .rs.getShinyFunction(params$name, params$where) : 
  attempt to apply non-function

In order to avoid above issue, I tries following options:

1. Load the .RData in Global environment.
        
            load(infile$datapath,.GlobalEnv)

2. Load the .RData in New Environment.

    ```r
        LoadToEnvironment <- function(RData, env = new.env())       { load(RData, env)  
         return(env) }  
       e <- LoadToEnvironment("D:\\Demo NBA\\AddOnPropensity.R")  
      val_modtyp <- e$val_modtyp  
      val_model <- e$val_model  
      val_b <-  e$val_b 
    ```

3. Load data Using attach(<filename>) 

Code Snippet:

```r
require(shinydashboard)||install.packages("shinydashboard"); library(shinydashboard)  
require(shiny)||install.packages("shiny"); library(shiny)  
require(shinyjs)||install.packages("shinyjs"); library(shinyjs) 


ui <- fluidPage(
  useShinyjs(),
  extendShinyjs(text = "shinyjs.refresh = function() { location.reload(); }"),
    fluidPage(id="Q1",useShinyjs(),
            
            dashboardPage(dashboardHeader(title="Test",titleWidth=400),
                          dashboardSidebar(),
                                            
                          dashboardBody (
                            
                              
                              tabItem("PMData",
                                      fileInput('filepm', 'Choose Data to Upload',accept = c(".R")),
                                      uiOutput('ui.PM2'),
                                      actionButton("savepm","Save"),
                                      uiOutput("ui.PM3")
                              )
            )
  )
  
  ))

server <- function(session,input,output){
  
  hide("savepm")
  
  dfPM <<- data.frame(Category=character(),
                      PredictiveModel=character(),
                      OfferIdentifier=character(),
                      stringsAsFactors=FALSE)
 
  

  
  LoadToEnvironment <<- function(RData, env = new.env())
  {  
    load(RData, env)  
    return(env) }
  
  val_choices <- reactive({
    if (is.null(input$filepm)){
      return()      
    }
    infile <- input$filepm    
    e <- LoadToEnvironment(infile$datapath)
    e$val_b
  }) 
  
 
  
  observeEvent(input$filepm,{
    
    useShinyjs()
    if (is.null(input$filepm)){
      return()      
    }
    
    output$ui.PM2 <- renderUI ({
      selectInput("offered",label= "Offered Test",choices = val_choices(),
                  selected = NULL)
    })
    show("ui.PM2")
    show("savepm")
  })
  
  val_pmfile <- reactive({
    if (is.null(input$filepm)){
      return()      
    }
    infile <- input$filepm
    infile$datapath
  }) 
  
  TempPredmodDF <- reactive({
    if(is.null(input$offered))
    {
      return()
    }else{
      data.frame(Category="Test",
                 PredictiveModel=val_pmfile(),
                 OfferIdentifier=input$offered,
                 stringsAsFactors=FALSE)}
  })
  
  observeEvent(input$savepm,
{  
  useShinyjs()
  
  tempPMdf <- TempPredmodDF()
  
  
  if(nrow(dfPM[dfPM$Category==tempPMdf$Category,]) == 0)
  {
    dfPM <<- rbind(dfPM,tempPMdf)
  }else
  {
    getidx <- as.numeric(which( dfPM[,1] == tempPMdf$Category ))
    
    dfPM[getidx,2] <<- tempPMdf$PredictiveModel
    
    dfPM[getidx,3] <<- tempPMdf$OfferIdentifier
    
  }
  
  output$ui.PM3 <- renderTable({
    dfPM},include.rownames=FALSE)
  
  hide("ui.PM2")
  hide("savepm")
  show("ui.PM3")
  
})
 
}


app <- shinyApp(ui,server)

runApp(app,port = 7000,launch.browser = getOption("shiny.launch.browser", interactive()))
```

**Comment by yihui** [comment]
Could you make your example self-contained? Without the .RData, I don't think we can reproduce your problem. It will help a lot if you can reduce your example, e.g. if shinyjs and shinydashboard are irrelevant, you may remove them and the corresponding code from the app. A minimal example will make it much easier to diagnose the issue. BTW, what is your RStudio version? This sounds like an RStudio IDE issue anyway.

**Comment by NxgChirag** [comment]
I have put the .Rdata file,which i used to upload. Please check it google link below. This will better help you to reproduce the problem.

groups.google.com/forum/#!topic/shiny-discuss/WUVlG4IZADM 

I am using RStudio Version 0.98.1062.

Simplified Code:


require(shiny)||install.packages("shiny"); library(shiny)  



ui <- fluidPage(fileInput('filepm', 'Choose Data to Upload',accept = c(".R")),
                uiOutput('ui.PM2'),
                actionButton("savepm","Save"),
                uiOutput("ui.PM3"))                          
           

server <- function(session,input,output){
  
  dfPM <<- data.frame(Category=character(),
                      PredictiveModel=character(),
                      OfferIdentifier=character(),
                      stringsAsFactors=FALSE)
  
  LoadToEnvironment <<- function(RData, env = new.env())
  {  
    load(RData, env)  
    return(env) }
  
  val_choices <- reactive({
    if (is.null(input$filepm)){
      return()      
    }
    infile <- input$filepm    
    e <- LoadToEnvironment(infile$datapath)
    e$val_b
  }) 
  
  observeEvent(input$filepm,{    
 
    if (is.null(input$filepm)){
      return()      
    }
    
    output$ui.PM2 <- renderUI ({
      selectInput("offered",label= "Offered Test",choices = val_choices(),
                  selected = NULL)
    })

  })
  
  val_pmfile <- reactive({
    if (is.null(input$filepm)){
      return()      
    }
    infile <- input$filepm
    infile$datapath
  }) 
  
  TempPredmodDF <- reactive({
    if(is.null(input$offered))
    {
      return()
    }else{
      data.frame(Category="Test",
                 PredictiveModel=val_pmfile(),
                 OfferIdentifier=input$offered,
                 stringsAsFactors=FALSE)}
  })
  
  observeEvent(input$savepm,
  {  
   
   tempPMdf <- TempPredmodDF()
  
  
    if(nrow(dfPM[dfPM$Category==tempPMdf$Category,]) == 0)
    {
      dfPM <<- rbind(dfPM,tempPMdf)
    }else
    {
      getidx <- as.numeric(which( dfPM[,1] == tempPMdf$Category ))    
      dfPM[getidx,2] <<- tempPMdf$PredictiveModel    
      dfPM[getidx,3] <<- tempPMdf$OfferIdentifier    
  }
  
  output$ui.PM3 <- renderTable({
    dfPM},include.rownames=FALSE) 
  
})

}

app <- shinyApp(ui,server)

runApp(app,port = 7000,launch.browser = getOption("shiny.launch.browser", interactive()))

**Comment by yihui** [comment]
Alright. Do you mind a blind try to upgrade your RStudio first? https://www.rstudio.com/products/rstudio/download/

**Comment by NxgChirag** [comment]
I tried with Version 0.99.669. Issue persists.

**Comment by yihui** [comment]
This is not reproducible to me. Please provide your session info:

```r
library(shiny)
devtools::session_info()
```

**Comment by NxgChirag** [comment]
Issue is resolved after R version upgraded from 3.1.2 to 3.2.1.
